rasa/cli/data.py

```diff
@@ -234,6 +273,422 @@ def split_nlu_data(args: argparse.Namespace) -> None:
+            )
+            rounded_augmented = round(
+                intent_report[intent_key][metric], significant_figures
+            )
+            if rounded_original != rounded_augmented:
+                changed_intents.add(intent_key)
+
+    return changed_intents
+
+
+def _create_augmentation_summary(
+    pooled_intents: Set,
+    changed_intents: Set,
+    classification_report: Dict[Text, Dict[Text, float]],
+    intent_report: Dict[Text, float],
+) -> Tuple[
+    Dict[Text, Dict[Text, float]], Dict[Text, float],
+]:
+
+    intent_summary = collections.defaultdict(dict)
+    for intent in (
+        pooled_intents | changed_intents | {"micro avg", "macro avg", "weighted avg"}
+    ):
+        if intent not in classification_report.keys():
+            continue
+
+        intent_results_original = classification_report[intent]
+        intent_results = intent_report[intent]
+
+        # Record performance changes for augmentation based on the diversity criterion
+        precision_change = (
+            intent_results["precision"] - intent_results_original["precision"]
+        )
+        recall_change = intent_results["recall"] - intent_results_original["recall"]
+        f1_change = intent_results["f1-score"] - intent_results_original["f1-score"]
+
+        intent_results["precision_change"] = intent_summary[intent][
+            "precision_change"
+        ] = precision_change
+        intent_results["recall_change"] = intent_summary[intent][
+            "recall_change"
+        ] = recall_change
+        intent_results["f1-score_change"] = intent_summary[intent][
+            "f1-score_change"
+        ] = f1_change
+        intent_report[intent] = intent_results
+
+    return (intent_summary, intent_report)
+
+
+def _plot_summary_reports(
+    intent_summary_diverse: Dict[Text, Dict[Text, float]],
+    intent_summary_random: Dict[Text, Dict[Text, float]],
+    changed_intents_diverse: Set[Text],
+    changed_intents_random: Set[Text],
+    output_directory_diverse: Text,
+    output_directory_random: Text,
+):
+
+    for metric in ["precision", "recall", "f1-score"]:
+        output_file_diverse = os.path.join(
+            output_directory_diverse, f"{metric}_changes.png"
+        )
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary_diverse,
+            changed_intents=changed_intents_diverse,
+            metric=metric,
+            output_file=output_file_diverse,
+        )
+
+        output_file_random = os.path.join(
+            output_directory_random, f"{metric}_changes.png"
+        )
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary_random,
+            changed_intents=changed_intents_random,
+            metric=metric,
+            output_file=output_file_random,
+        )
+
+
+def suggest_nlu_data(args: argparse.Namespace) -> None:
```

Comment:
Unfortunately this function is too long, currently sitting at ~220 lines of code. We should split it into separate parts - following a similar length to other functions in this PR would be a good starting point!